src/NuGet.Core/NuGet.ProjectModel/LockFile/LockFileTargetLibrary.cs

```diff
@@ -132,6 +139,16 @@ public override int GetHashCode()
                 combiner.AddObject(item);
             }
 
+            foreach (var item in ToolsAssemblies.OrderBy(item => item.Path, StringComparer.OrdinalIgnoreCase))
```

Comment:
If case does matter then then use Ordinal sort, if not I would expect it to hash with ignore case

Reply:
The "item" element is LockFileItem....the lockFileItem hashcode does  the following:
```
        public override int GetHashCode()
        {
            var combiner = new HashCodeCombiner();

            combiner.AddStringIgnoreCase(Path);
            combiner.AddDictionary(Properties);

            return combiner.CombinedHash;
        }
```